### PR TITLE
Item fixes and World fixes

### PR DESCRIPTION
**Pull request by BoogieMonster1O1**
very important stuff here
around 40% of methods are mapped now
much better than before

### CODE REVIEW

**Comment by BoogieMonster1O1** [comment]
unnecessary changelog 
```diff
 On branch unc0ver1
 Changes to be committed:
-	deleted:    mappings/net/minecraft/GrassColorResourceReloadListener.mapping
+	modified:   mappings/net/minecraft/block/Block.mapping
-	deleted:    mappings/net/minecraft/block/Fertilizable.mapping
+	new file:   mappings/net/minecraft/block/Growable.mapping
+	modified:   mappings/net/minecraft/block/entity/SpawnerBlockEntityBehavior.mapping
-	deleted:    mappings/net/minecraft/block/item/PillarItem.mapping
+	new file:   mappings/net/minecraft/class_427.mapping
+	modified:   mappings/net/minecraft/client/render/model/WeightedBakedModel.mapping
+	modified:   mappings/net/minecraft/enchantment/EnchantmentHelper.mapping
-	renamed:    mappings/net/minecraft/enchantment/EnchantmentInfo.mapping -> mappings/net/minecraft/enchantment/EnchantmentInfoEntry.mapping
+	modified:   mappings/net/minecraft/entity/Entity.mapping
+	modified:   mappings/net/minecraft/entity/EntityType.mapping
+	modified:   mappings/net/minecraft/entity/LivingEntity.mapping
+	modified:   mappings/net/minecraft/entity/decoration/DecorationEntity.mapping
+	modified:   mappings/net/minecraft/entity/player/HungerManager.mapping
+	modified:   mappings/net/minecraft/entity/player/PlayerEntity.mapping
+	modified:   mappings/net/minecraft/entity/player/PlayerInventory.mapping
+	modified:   mappings/net/minecraft/entity/projectile/FishingBobberEntity.mapping
+	modified:   mappings/net/minecraft/entity/vehicle/AbstractMinecartEntity.mapping
+	modified:   mappings/net/minecraft/item/ArmorItem.mapping
+	modified:   mappings/net/minecraft/item/ArmorStandItem.mapping
+	modified:   mappings/net/minecraft/item/AxeItem.mapping
+	modified:   mappings/net/minecraft/item/BlockItem.mapping
+	modified:   mappings/net/minecraft/item/BucketItem.mapping
+	new file:   mappings/net/minecraft/item/CropItem.mapping
+	modified:   mappings/net/minecraft/item/DyeItem.mapping
+	modified:   mappings/net/minecraft/item/EnchantedBookItem.mapping
+	modified:   mappings/net/minecraft/item/FilledMapItem.mapping
+	modified:   mappings/net/minecraft/item/FireworkChargeItem.mapping
+	modified:   mappings/net/minecraft/item/FoodItem.mapping
+	modified:   mappings/net/minecraft/item/GrassBlockItem.mapping
+	new file:   mappings/net/minecraft/item/HoeItem.mapping
+	modified:   mappings/net/minecraft/item/Item.mapping
-	deleted:    mappings/net/minecraft/item/ItemFrameItem.mapping
+	modified:   mappings/net/minecraft/item/ItemStack.mapping
+	new file:   mappings/net/minecraft/item/LilyPadItem.mapping
+	modified:   mappings/net/minecraft/item/MinecartItem.mapping
+	new file:   mappings/net/minecraft/item/MusicDiscItem.mapping
+	modified:   mappings/net/minecraft/item/PickaxeItem.mapping
+	new file:   mappings/net/minecraft/item/PlaceableItem.mapping
-	deleted:    mappings/net/minecraft/item/PotatoItem.mapping
-	deleted:    mappings/net/minecraft/item/RecordItem.mapping
-	deleted:    mappings/net/minecraft/item/StoneSlabBlockItem.mapping
+	new file:   mappings/net/minecraft/item/StoneSlabItem.mapping
-	deleted:    mappings/net/minecraft/item/StringItem.mapping
+	modified:   mappings/net/minecraft/item/SwordItem.mapping
+	modified:   mappings/net/minecraft/item/VariantBlockItem.mapping
+	new file:   mappings/net/minecraft/item/WallHangableItem.mapping
-	deleted:    mappings/net/minecraft/item/WaterLilyItem.mapping
+	modified:   mappings/net/minecraft/item/WoodenDoorItem.mapping
+	modified:   mappings/net/minecraft/item/WrittenBookItem.mapping
+	modified:   mappings/net/minecraft/item/map/MapState.mapping
-	renamed:    mappings/net/minecraft/nbt/NbtTag.mapping -> mappings/net/minecraft/nbt/Tag.mapping
+	new file:   mappings/net/minecraft/resource/GrassColorResourceReloadListener.mapping
-	renamed:    mappings/net/minecraft/structure/Structure.mapping -> mappings/net/minecraft/structure/StructureGenerator.mapping
+	new file:   mappings/net/minecraft/util/ChatSerializer.mapping
*	renamed:    mappings/net/minecraft/class_1712.mapping -> mappings/net/minecraft/util/PlayerSelector.mapping
+	new file:   mappings/net/minecraft/util/WeightedRandomChestContent.mapping
+	new file:   mappings/net/minecraft/util/WeightedRandomFishingLoot.mapping
+	modified:   mappings/net/minecraft/util/math/ColumnPos.mapping
+	modified:   mappings/net/minecraft/world/World.mapping
+	modified:   mappings/net/minecraft/world/chunk/ChunkProvider.mapping
-	renamed:    mappings/net/minecraft/client/particle/ParticleTypes.mapping -> mappings/net/minecraft/client/particle/ParticleType.mapping
+	new file:   mappings/net/minecraft/client/world/ChunkRenderCache.mapping
+	modified:   mappings/net/minecraft/util/math/BlockBox.mapping
+	modified:   mappings/net/minecraft/world/World.mapping
+	modified:   mappings/net/minecraft/world/WorldView.mapping
+	modified:   mappings/net/minecraft/world/chunk/Chunk.mapping
+	modified:   mappings/net/minecraft/world/chunk/ChunkProvider.mapping
```

File: mappings/net/minecraft/block/Growable.mapping [hunk 1]
```diff
@@ -0,0 +1,16 @@
+CLASS net/minecraft/class_164 net/minecraft/block/Growable
+	METHOD method_750 canGrow (Lnet/minecraft/class_99;Lnet/minecraft/class_1372;Lnet/minecraft/class_376;Z)Z
+		ARG 1 world
+		ARG 2 pos
+		ARG 3 state
+		ARG 4 bl
+	METHOD method_751 canGrow (Lnet/minecraft/class_99;Ljava/util/Random;Lnet/minecraft/class_1372;Lnet/minecraft/class_376;)Z
```
Comment (hYdos) [inline mappings/net/minecraft/block/Growable.mapping:7]: duplicate method

Comment (BoogieMonster1O1) [reply mappings/net/minecraft/block/Growable.mapping]: fixed

File: mappings/net/minecraft/item/ItemStack.mapping [hunk 2]
```diff
@@ -39,12 +45,15 @@ CLASS net/minecraft/class_2056 net/minecraft/item/ItemStack
...
 		ARG 4 selected
-	METHOD method_8339 (Lnet/minecraft/class_99;Lnet/minecraft/class_1963;)Lnet/minecraft/class_2056;
+	METHOD method_8339 onStartUse (Lnet/minecraft/class_99;Lnet/minecraft/class_1963;)Lnet/minecraft/class_2056;
+		ARG 1 world
 		ARG 2 player
 	METHOD method_8340 onCraft (Lnet/minecraft/class_99;Lnet/minecraft/class_1963;I)V
 		ARG 1 world
 		ARG 2 player
 		ARG 3 amount
+	METHOD method_8341 getMiningSpeedForBlock (Lnet/minecraft/class_160;)F
```
Comment (shedaniel) [inline mappings/net/minecraft/item/ItemStack.mapping:55]: getMiningSpeedMultiplier

File: mappings/net/minecraft/item/PickaxeItem.mapping [hunk 3]
```diff
@@ -1,2 +1,4 @@
 CLASS net/minecraft/class_9 net/minecraft/item/PickaxeItem
 	FIELD field_38 EFFECTIVE_BLOCKS Ljava/util/Set;
+	METHOD <init> (Lnet/minecraft/class_2054$class_2055;)V
+		ARG 1 nmaterial
```
Comment (shedaniel) [inline mappings/net/minecraft/item/PickaxeItem.mapping:4]: typo

**Pull request merged**